functests/gatekeeper/gatekeeper.go

```diff
@@ -0,0 +1,151 @@
+package gatekeeper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	dynamic "github.com/openshift-kni/cnf-features-deploy/functests/utils/dynamic"
+)
+
+type metadataInfo struct {
+	kind  string
+	key   string
+	value string
+}
+
+type assignMetadataInfo struct {
+	name     string
+	kinds    []string
+	scope    string
+	metadata metadataInfo
+}
+
+var _ = Describe("gatekeeper", func() {
+	// we make the interface key to a var for runtime key check
+	// if we used the string directly a typo will lead to a runtime error
+	// typos in var names will make a compile error
+	assignMetadata := "assignmetadata"
+	pods := "pods"
+	// create the dynamic client
+	client := dynamic.New("")
+	// create a gvr list for to resources we want an interface with
+	gvrList := []schema.GroupVersionResource{
+		{
+			Group:    "mutations.gatekeeper.sh",
+			Version:  "v1alpha1",
+			Resource: assignMetadata,
+		},
+		{
+			Group:    "",
+			Version:  "v1",
+			Resource: pods,
+		},
+	}
+	// config the gvrList on the dynamic client
+	client.ConfigGVR(gvrList...)
+	// dynamic client.Assert is used to add safety without the need to check on every use
+	Expect(client.Assert(assignMetadata)).To(Equal(true))
+	Expect(client.Assert(pods)).To(Equal(true))
+
+	DescribeTable("Add Metadata info(labels/annotations) to resource",
+		func(amInfo assignMetadataInfo, resource *unstructured.Unstructured) {
+			amDefinition := assignMetadataDefinition(&amInfo)
+
+			_, err := client.Resource(assignMetadata).Create(context.Background(), amDefinition, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			time.Sleep(1 * time.Second)
+
+			r, err := client.Resource(resource.GetKind()).Create(context.Background(), resource, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			value, ok, err := unstructured.NestedStringMap(r.Object, "metadata", amInfo.metadata.kind, "mutated")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ok).To(BeTrue())
+			Expect(value).To(Equal("true"))
+		},
+		Entry(
+			"pod-mutation",
+			assignMetadataInfo{
+				name:  "test-pod-mutaion",
+				kinds: []string{"pods"},
+				scope: "namespaced",
+				metadata: metadataInfo{
+					kind:  "labels",
+					key:   "mutated",
+					value: "true",
+				},
+			},
+		),
+	)
+
+	DescribeTable("Do not mutate existing metadata info",
+		func(amInfo assignMetadataInfo, resource *unstructured.Unstructured) {
+			amDefinition := assignMetadataDefinition(&amInfo)
+
+			_, err := client.Resource(assignMetadata).Create(context.Background(), amDefinition, metav1.CreateOptions{})
```

Comment:
`client.resource[assignMetadata]` already contains this client. Either use that one, or don't precreate them in the client.